### DISCUSSION TITLE
Some thoughts on Appium2.0 migrate doc

### DISCUSSION THREAD

**Discussion by youngfreeFJS** [appium/appium#15801] [General]
Some thoughts on [Appium2.0 migrate doc ](https://github.com/appium/appium/blob/2.0/docs/en/advanced-concepts/migrating-to-appium-2.0.md#bangbang-installing-drivers-during-setup)
1、how to develop an appium-plugin ?
2、As a plugin owner, should I maintained the plugin-client ?
3、Is there a sample plugin that explains how to use it？
4、Can I use appium2.0  with appium/java-client latest now?

Thanks for any reply！

**Comment by jlipps** [comment]
The Appium core plugins repo is [here](https://github.com/appium/appium-plugins). If you look at the [base plugin](https://github.com/appium/appium-plugins/tree/master/packages/base), you will see documentation on how to create your own. Looking at the other plugins in the repo will be a good start.

And I believe the latest java-client supports Appium 2.0, yes.

**Reply by KazuCocoa** [reply]
Yes, selenium 3 based clients should work.
One note about client side is that `forceMjsonwp` no longer works. It drops W3C capabilities on client side, so Appium 2.0 cannot handle the request.
